starfish/pipeline/features/spots/detector/gaussian.py

```diff
@@ -17,7 +17,8 @@ class GaussianSpotDetector(SpotFinderAlgorithmBase):
 
```

Comment:
`skimage` recently added 3d support for blob_log, so I've used that to generalize this approach to 2d and 3d. This allowed me to simplify some of the code. I've also added some missing type hints and documentation.